operator/internal/status/lokistack.go

```diff
@@ -74,6 +75,16 @@ func SetDegradedCondition(ctx context.Context, k k8s.Client, req ctrl.Request, m
 	return updateCondition(ctx, k, req, degraded)
 }
 
+func SetWarningCondition(ctx context.Context, k k8s.Client, req ctrl.Request, msg string, reason lokiv1.LokiStackConditionReason) error {
```

Comment:
This does not really work as I expected. The warning conditions need to be able to be active in addition to the other conditions. I think the overall "health" of the LokiStack should still be "Ready" even when there are warning conditions active. This would also leave us the option of having more than one warning condition active at the same time, which is already possible with the two warnings we currently have.